### DISCUSSION TITLE
Tile inspector function: Copy Tile

### DISCUSSION THREAD

**Discussion by WackedSwe** [OpenRCT2/OpenRCT2#15772] [Feature requests]
As simple as it sounds ( and I gave the search function a looksie but to no avail ) 

We'd love to see a "copy tile" function, for fences, walls. In to iterations

1: Mark what you want and click "copy" (skip the surface layer)
2: A tickbox or icon for "copy full tile"   (skip the surface layer)

This would be an enchancement as it'd save a lot of people the fuss of creating structures by saving rides with scenery. This would give us the option to have a corner of a park with items to copy. 

First requested with perhaps better wording in #1926 . The ghosting for me is not a needed feature, but itd help immensly.

**Comment by AaronVanGeffen** [comment]
Aside from the technical implementation, this requires a rethink about what the tile inspector toolbar should look like. I would be against simply adding another copy and paste button…

For discussion purposes, this is the current state of the window's toolbar:

![Screenshot_20200205_003051](https://user-images.githubusercontent.com/604665/73796919-e334ee80-47ae-11ea-9134-a4cc55f4a57c.png)

**Comment by ocalhoun6** [comment]
If we're going to do a copy/paste function, I'd rather do it all the way to begin with. A totally new tool, separate from the tile inspector. 

1: A main window with options for 'select area to copy' and 'paste'. (With paste greyed out if there are no saved selections.)

2: A selection window that lets you select an arbitrary rectangle up to a given size, and lets you turn on or off selecting the underlying surface/water elements.

3: A save dialog that lets you assign a filename for your saved scenery group (saved in a new format unique to OpenRCT2, since there's no question of backward compatibility.)

4: A paste window that lets you select a file from the scenery groups folder, and will give you a warning if it includes any scenery items not in the currently loaded map ... maybe with a convenient 'load missing items' button available. Once selected, you'd then get a ghost preview of your scenery group you could place like a saved coaster design, with the ability to mirror, rotate, and raise/lower it.

------------------------------

It would be a lot more work, perhaps, than this 1-tile tile inspector stuff, but it would also be a lot more powerful. Great for duplicating repetitious scenery, great for creating perfectly symmetrical scenery, and really amazing for taking big scenery projects from one park and dropping them into a different park. Could even be really useful just for copying/pasting terrain elements ... like if you made a really good mountain one time and just want to copy/paste it into a new park.

Probably best not to include any trackitecture/rides in the selection, at least at first, leaving it scenery only. Pasting multiple rides or partial rides might have a lot of issues complicating the matter more than it needs to be.

Anyway, that's my opinion -- if you're going to build a real scenery copy/paster, do it right the first time, rather than needing to rebuild it to add better features later on.

**Comment by WackedSwe** [comment]
How about a simple addition to the already working copy-paste icon for time being? Should not be impossible for it to handle two+ objects.  I agree that the game lacks cloning, but most want it to be simple and handy. A code looksie to see if it is possible is simpler than redrawing something that can take months =)

**Comment by duncanspumpkin** [comment]
Feel free to implement it and put up a PR.

**Comment by J-PIE-314** [comment]
This plugin for OpenRCT2 can do what you are requesting here: https://github.com/Sadret/openrct2-scenery-manager
